pkg/controllers/elasticsearch/role/resources.go

```diff
@@ -0,0 +1,33 @@
+package role
+
+import (
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/jetstack/navigator/pkg/apis/navigator"
+	"github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
+	"github.com/jetstack/navigator/pkg/controllers/elasticsearch/util"
+)
+
+func roleForCluster(c *v1alpha1.ElasticsearchCluster) *rbacv1beta1.Role {
+	return &rbacv1beta1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            util.RBACRoleName(c),
+			Namespace:       c.Namespace,
+			OwnerReferences: []metav1.OwnerReference{util.NewControllerRef(c)},
+			Labels:          util.ClusterLabels(c),
+		},
+		Rules: []rbacv1beta1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Verbs:     []string{"create", "update", "patch"},
+				Resources: []string{"events"},
+			},
+			{
+				APIGroups: []string{navigator.GroupName},
+				Verbs:     []string{"update", "get"},
+				Resources: []string{"pilots", "elasticsearchclusters"},
+			},
+		},
+	}
+}
```

Comment:
❓ Maybe this function should be called `pilotRoleForCluster` (or something less generic)

Reply:
This isn't generic between controllers - cassandra pilots shouldn't be able to access elasticsearchclusters. I'd also like to keep it quite explicit and obvious what we're granting, at least for now.